src/web/mjs/connectors/MangaHost.mjs

```diff
@@ -28,73 +39,50 @@ export default class MangaHost extends Connector {
         return new Manga(this, uri.pathname, data[0].textContent.trim());
     }
 
-    _getMangaListFromPages(mangaPageLinks, index) {
+    async _getMangaListFromPages(mangaPageLinks, index) {
         if (index === undefined) {
             index = 0;
         }
-        return this.wait(0)
-            .then(() => this.fetchDOM(mangaPageLinks[index], 'a.manga-block-title-link', 5))
-            .then(data => {
-                let mangaList = data.map(element => {
-                    return {
-                        id: this.getRelativeLink(element),
-                        title: element.text.trim()
-                    };
-                });
-                if (index < mangaPageLinks.length - 1) {
-                    return this._getMangaListFromPages(mangaPageLinks, index + 1)
-                        .then(mangas => mangas.concat(mangaList));
-                } else {
-                    return Promise.resolve(mangaList);
-                }
-            });
+        await this.wait(this.config.throttle.value);
+        const data = await this.fetchDOM(mangaPageLinks[index], 'a.manga-block-title-link', 5);
+
+        let mangaList = data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
+        if (index < mangaPageLinks.length - 1) {
+            return this._getMangaListFromPages(mangaPageLinks, index + 1)
```

Comment:
To prevent `Maximum call stack size exceeded` error, it is recommend to use a [loop](https://github.com/manga-download/hakuneko/blob/master/src/web/mjs/connectors/templates/%40Samples.mjs#L50) for too many calls.

Reply:
original code :\ didn't rework much. I guess i'll have to clean someone else code :'(